sources/packages/web/src/composables/useAuth.ts

```diff
@@ -1,10 +1,28 @@
-import { AppConfigService } from "@/services/AppConfigService";
-import { computed } from "vue";
-
-export function useAuth() {
-  const isAuthenticated = computed(
-    () => AppConfigService.shared.authService?.authenticated === true,
-  );
-
-  return { isAuthenticated };
-}
+import { RouteHelper } from "@/helpers";
+import { AuthService } from "@/services/AuthService";
+import { AppIDPType, ClientIdType } from "@/types";
+import { computed } from "vue";
+
+export function useAuth() {
+  const isAuthenticated = computed(
+    () => AuthService.shared.keycloak?.authenticated === true,
+  );
+
+  const parsedToken = computed(() => AuthService.shared.userToken);
+
+  const executeLogin = async (
+    clientType: ClientIdType,
+    idp: AppIDPType,
+  ): Promise<void> => {
+    return AuthService.shared.keycloak?.login({

```

Comment:
the promise is `void` but we are returning

Reply:
As per our conversation, IMO, returning void is different from returning a Promise of something. Both ways are going to work, but l will change to use an await instead because maybe it could avoid confusion in the future.

Reply:
Changed to use await.